services/src/main/java/org/keycloak/userprofile/DeclarativeUserProfileProvider.java

```diff
@@ -255,8 +257,17 @@ protected UserProfileMetadata decorateUserProfileForCache(UserProfileMetadata de
             return decoratedMetadata;
         }
 
+        decorateWithAttributes(decoratedMetadata, context, parsedConfig.getAttributes(), asHashMap(parsedConfig.getGroups()));
+        return decoratedMetadata;
+    }
+
+    private Map<String, UPGroup> asHashMap(List<UPGroup> groups) {
+        return groups.stream().collect(Collectors.toMap(g -> g.getName(), g -> g));
+    }
+
+    private void decorateWithAttributes(UserProfileMetadata decoratedMetadata, UserProfileContext context, List<UPAttribute> attributes, Map<String, UPGroup> groupsByName) {
```

Comment:
I do not see any need for new method here, simply take groups from the config and use them in attributes config loop inside of the `decorateUserProfileForCache()` method